src/main/kotlin/no/skatteetaten/aurora/gobo/integration/unclematt/ProbeService.kt

```diff
@@ -0,0 +1,54 @@
+package no.skatteetaten.aurora.gobo.integration.unclematt
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.skatteetaten.aurora.gobo.ServiceTypes
+import no.skatteetaten.aurora.gobo.TargetService
+import no.skatteetaten.aurora.gobo.integration.SourceSystemException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.reactive.function.client.bodyToMono
+
+@Service
+class ProbeService(
+    @TargetService(ServiceTypes.UNCLEMATT) val webClient: WebClient
+) {
+    private val logger: Logger = LoggerFactory.getLogger(ProbeService::class.java)
+    private val objectMapper = jacksonObjectMapper()
+    private val exceptionMsg = "Failed to invoke firewall scan in the cluster"
+    private val exceptionUserMsg = "Failed to perform netdebug operation"
+
+    init {
+        objectMapper.enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE)
+    }
+
+    fun probeFirewall(host: String, port: Int): List<ProbeResult> {
+        try {
+            val response = webClient
+                .get()
+                .uri {
+                    it.path("/scan/v1")
+                        .queryParam("host", host)
+                        .queryParam("port", port)
+                        .build()
+                }
+                .retrieve()
+                .bodyToMono<String>()
+                .block() ?: return emptyList()
+
+            val probeResultList = objectMapper.readValue<List<ProbeResult>>(response)
```

Comment:
Du kan skrive List<ProbeResult> på linje 40 i steden for string, da slipper du å unmarshalle 2 ganger.

Reply:
Som vi pratet om, så kan denne kommentaren ignoreres siden vi trenger en custom objectMapper.

Reply:
Evt kan man se her hvordan vi gjør det i boober client.
https://github.com/Skatteetaten/gobo/pull/13/files

Reply:
Da har jeg endret det til at WebClient for UncleMatt får tilordnet ObjectMapper via ExchangeStrategies.

Problemet nå er mye gjentakende kode i ApplicationConfig.